File: BE/src/main/java/com/ssafy/tax7i/card/dto/CardResponse.java
package com.ssafy.tax7i.card.dto;

import com.ssafy.tax7i.card.entity.Card;
import com.ssafy.tax7i.card.entity.CardType;
import java.time.LocalDateTime;

public record CardResponse(
        Long id,
        String cardName,
        CardType cardType,
        String last4Digits,
        Boolean isDefault,
        String cardExpiryDate,
        String withdrawalDate,
        LocalDateTime createdAt
) {
    public static CardResponse from(Card card) {
        return new CardResponse(
                card.getId(),
                card.getCardName(),
                card.getCardType(),
                card.getLast4Digits(),
                card.isDefault(),
                card.getCardExpiryDate(),
                card.getWithdrawalDate(),
                card.getCreatedAt()
        );
    }
}

File: FE/app/src/main/java/com/ssafy/seveniTax/data/model/card/CardResponse.kt
package com.ssafy.seveniTax.data.model.card

data class CardResponse(
    val id: Long,
    val cardName: String,
    val cardType: String,
    val last4Digits: String,
    val isDefault: Boolean,
    val cardExpiryDate: String?,
    val withdrawalDate: String?,
    val createdAt: String?
)
